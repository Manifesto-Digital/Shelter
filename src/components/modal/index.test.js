import React from 'react';
import { act } from 'react-dom/test-utils';
import ReactModal from 'react-modal';
import Modal from '.';
import { mountWithTheme } from 'test-helpers';

const getCss = node =>
  window
    .getComputedStyle(node)
    .cssText.split(';')
    .join(';\n');

it('renders correct styles', () => {
  const wrapper = mountWithTheme(<Modal isOpen />);

  const { overlay, content } = wrapper.find(ReactModal).instance().portal;

  expect(getCss(overlay)).toMatchSnapshot('overlay');
  expect(getCss(content)).toMatchSnapshot('content');
});

it('requests to close when clicking the close button', () => {
  const onRequestClose = jest.fn();
  const wrapper = mountWithTheme(
    <Modal isOpen onRequestClose={onRequestClose} />
  );

  act(() => {
    wrapper
      .find(ReactModal)
      .find('button[title="Close"]')
      .simulate('click');
  });

  expect(onRequestClose).toHaveBeenCalledTimes(1);
});
