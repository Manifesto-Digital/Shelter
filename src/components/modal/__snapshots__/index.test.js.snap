// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correct styles: content 1`] = `
"display: block;
 position: relative;
 bottom: 40px;
 background: rgb(255, 255, 255);
 overflow: auto;
 border-radius: none;
 outline: none;
 padding: 40px;
 background-color: rgb(247, 247, 247);
"
`;

exports[`renders correct styles: overlay 1`] = `
"display: flex;
 position: fixed;
 top: 0px;
 left: 0px;
 right: 0px;
 bottom: 0px;
 background-color: rgba(43, 46, 56, 0.9);
 align-items: center;
 justify-content: center;
"
`;
